### ISSUE TITLE
Support of "StaticBlocks, Private Identifier in "in" check & "Import Assertions"

### ISSUE THREAD

**Issue by jogibear9988** [sebastienros/esprima-dotnet#219]
Hello,

I'm currently working on bring support for them into esprima-next. (hopefully the next few days.

@lahma 
Can you then check my pull against "esprima-next" when it's done, so later the implementation against esprima-net could be nearly the same?

Hopefully after that I find time for the "refactor" of the Visitor.

**Comment by lahma** [comment]
Sure thing, just ping me when needed and I'll try to have a look.

**Comment by jogibear9988** [comment]
@lahma 
I'm already done with it... :-)

see: https://github.com/node-projects/esprima-next/pull/21

**Comment by jogibear9988** [comment]
I'm not realy interested to do pull requests against original esprima, cause there is really no development. And also the last release of their npm package it's from 2018

**Comment by lahma** [comment]
Didn't spot anything to complain about, a lot of files seems to be changed to some tree format change. The whitelist.txt diff seems to be quite good indicator how well things perform. There seems to be very few `assert` matches, but quite a lot of `private` matches, which seem to original from generator methods - which probably mean out of scope for the class support.

It might be easier to handle if import and class changes would be two separate things.

**Comment by jogibear9988** [comment]
I've seperated "Import" and "class" changes they are all in one commit:

class changes: https://github.com/node-projects/esprima-next/pull/21/commits/b6ec3c79d317ed0d8d861c72567730a0f723ba3e
import assertions: https://github.com/node-projects/esprima-next/pull/21/commits/a82b4abe61e48586fb39af7db4f4a72c4f800f4d
private in support : https://github.com/node-projects/esprima-next/pull/21/commits/3f4fa42bf6cf6e33851fbf4c13b142ed1e1c476a

the rest is working that testing works again, test-262 updates, switch to new test-262 version and so on...

**Comment by lahma** [comment]
Awesome, much easier to grok now! I have also now initial setup for esprima-dotnet that allows running the test262 test suite with same kind of allow-list setup that should make it easier to validate whether things work as expected.

**Comment by jogibear9988** [comment]
@lahma 

I now look to merge the "StaticBlock" support for classes.
But to implement it according "Estree" spec, StaticBlock should inherit from BlockStatement, see https://github.com/estree/estree/blob/master/es2022.md

Should I remove the Sealed of this?

**Comment by jogibear9988** [comment]
Maybe sealed is there as a Perf. optimization?

**Comment by lahma** [comment]
Things are sealed primarily for less worrying about whether someone has inherited (can we change this?) and secondarily for perf (seal everything until someone has a valid use case to unseal). If the spec says it's a is-a relationship I think it warrants for unsealing.

**Comment by jogibear9988** [comment]
@lahma should I also create a pull for import assertions, even if it's only a stage 3 proposal?

**Comment by lahma** [comment]
Maybe we should wait for stage 4 so that we know the spec is solid. Test262 suite will have test cases only after stage 4 is hit.
